src/sqlfluff/core/templaters/slicers/tracer.py

```diff
@@ -157,6 +170,12 @@ def record_trace(self, target_slice_length):
         if slice_type in ("literal", "templated"):
             self.source_idx += target_slice_length
 
+    def next_slice_id(self) -> str:
```

Comment:
This function replaces uses of `uuid.uuid4().hex`. There was really no need to use UUIDs, and changing to a deterministic counter makes the tests easier to understand and debug.